lib/Dialect/TTMetal/Transforms/Passes.cpp

```diff
@@ -16,44 +16,397 @@
 #include "ttmlir/Dialect/TTIR/IR/TTIR.h"
 #include "ttmlir/Dialect/TTIR/IR/TTIROps.h"
 #include "ttmlir/Dialect/TTIR/Transforms/Passes.h"
+#include "llvm/ADT/MapVector.h"
 
 #include "ttmlir/Dialect/TTKernel/IR/TTKernel.h"
 #include "ttmlir/Dialect/TTKernel/IR/TTKernelOps.h"
 #include "ttmlir/Dialect/TTKernel/IR/TTKernelOpsTypes.h"
 #include "ttmlir/Dialect/TTMetal/IR/TTMetalOpsTypes.h"
+#include "ttmlir/Utils.h"
+
+namespace mlir::tt::ttmetal {
```

Comment:
I suppose this struct would be useful even in TTNN backend, so maybe keep it in mlir::tt namespace and maybe move it to a place for common structs?